File: src/scripts/generate-icons.js
/**
 * Script para generar iconos de la PWA
 * Ejecutar con: node scripts/generate-icons.js
 * 
 * Nota: Si este script falla, usa el generador HTML en /public/icons/icon-generator.html
 * o genera los iconos manualmente con PWA Builder: https://www.pwabuilder.com/imageGenerator
 */

const fs = require('fs');
const path = require('path');

// Tamaños de iconos requeridos
const sizes = [72, 96, 128, 144, 152, 192, 384, 512];

// Crear directorio de iconos si no existe
const iconsDir = path.join(__dirname, '../public/icons');
if (!fs.existsSync(iconsDir)) {
  fs.mkdirSync(iconsDir, { recursive: true });
}

// Función para crear SVG de icono
function createIconSVG(size) {
  const padding = size * 0.15;
  const iconWidth = size - (padding * 2);
  const iconHeight = iconWidth * 0.75;
  const strokeWidth = size * 0.08;
  const circleRadius = size * 0.08;
  
  return `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${size}" height="${size}" viewBox="0 0 ${size} ${size}" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <linearGradient id="gradient" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#10b981;stop-opacity:1" />
      <stop offset="100%" style="stop-color:#059669;stop-opacity:1" />
    </linearGradient>
  </defs>
  
  <!-- Fondo con degradado -->
  <rect width="${size}" height="${size}" fill="url(#gradient)"/>
  
  <!-- Icono de tienda/carrito -->
  <g stroke="white" stroke-width="${strokeWidth}" stroke-linecap="round" stroke-linejoin="round" fill="none">
    <!-- Rectángulo principal -->
    <rect x="${padding}" y="${padding * 1.3}" width="${iconWidth}" height="${iconHeight}" rx="${size * 0.08}"/>
    
    <!-- Línea superior -->
    <line x1="${padding}" y1="${padding * 1.8}" x2="${size - padding}" y2="${padding * 1.8}"/>
    
    <!-- Handle superior -->
    <line x1="${size / 2}" y1="${padding * 1.3}" x2="${size / 2}" y2="${padding * 1.8}"/>
  </g>
  
  <!-- Círculos del carrito -->
  <circle cx="${padding + iconWidth * 0.3}" cy="${size - padding * 0.8}" r="${circleRadius}" fill="white"/>
  <circle cx="${padding + iconWidth * 0.7}" cy="${size - padding * 0.8}" r="${circleRadius}" fill="white"/>
  
  <!-- Texto pequeño para desarrollo -->
  ${size >= 192 ? `<text x="${size / 2}" y="${size * 0.55}" font-family="Arial" font-size="${size * 0.15}" fill="white" text-anchor="middle" font-weight="bold">GM</text>` : ''}
</svg>`;
}

console.log('🎨 Generando iconos de la PWA...\n');

// Generar iconos SVG (que servirán como placeholder hasta tener PNG reales)
sizes.forEach(size => {
  const svg = createIconSVG(size);
  const filename = `icon-${size}x${size}.svg`;
  const filepath = path.join(iconsDir, filename);
  
  fs.writeFileSync(filepath, svg);
  console.log(`✅ Generado: ${filename}`);
});

// Generar icono maskable
const maskableSVG = createIconSVG(512);
fs.writeFileSync(path.join(iconsDir, 'maskable-icon-512x512.svg'), maskableSVG);
console.log('✅ Generado: maskable-icon-512x512.svg');

console.log('\n⚠️  IMPORTANTE: Estos son archivos SVG temporales.');
console.log('📝 Para generar PNG reales, usa una de estas opciones:\n');
console.log('   1. Abre /public/icons/icon-generator.html en tu navegador');
console.log('   2. Usa PWA Builder: https://www.pwabuilder.com/imageGenerator');
console.log('   3. Convierte los SVG a PNG con una herramienta online\n');
console.log('✨ Los archivos SVG funcionarán temporalmente, pero PNG es mejor para PWAs.\n');

File: src/scripts/generate-icons.py
#!/usr/bin/env python3
"""
Script para generar iconos PWA desde el logo de Gualán Market

Requisitos:
    pip install Pillow

Uso:
    python3 scripts/generate-icons.py

El script buscará el logo en /public/logo-source.png y generará
los 9 iconos necesarios en /public/icons/
"""

import os
import sys
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("❌ ERROR: Pillow no está instalado")
    print("\n📦 Para instalar:")
    print("   pip install Pillow")
    print("   # o")
    print("   pip3 install Pillow")
    print("\n💡 O usa una alternativa:")
    print("   - Abre: public/icons/generate-icons-from-logo.html")
    print("   - O usa: https://www.pwabuilder.com/imageGenerator\n")
    sys.exit(1)

# Configuración
SCRIPT_DIR = Path(__file__).parent
PROJECT_ROOT = SCRIPT_DIR.parent
LOGO_PATH = PROJECT_ROOT / 'public' / 'logo-source.png'
OUTPUT_DIR = PROJECT_ROOT / 'public' / 'icons'

ICON_SIZES = [
    {'size': 72, 'name': 'icon-72x72.png'},
    {'size': 96, 'name': 'icon-96x96.png'},
    {'size': 128, 'name': 'icon-128x128.png'},
    {'size': 144, 'name': 'icon-144x144.png'},
    {'size': 152, 'name': 'icon-152x152.png'},
    {'size': 192, 'name': 'icon-192x192.png'},
    {'size': 384, 'name': 'icon-384x384.png'},
    {'size': 512, 'name': 'icon-512x512.png'},
    {'size': 512, 'name': 'maskable-icon-512x512.png', 'maskable': True}
]

def check_logo():
    """Verificar que existe el logo"""
    if not LOGO_PATH.exists():
        print(f"❌ ERROR: No se encontró el logo en: {LOGO_PATH}")
        print("\n📝 Para extraer el logo:")
        print("1. Corre el proyecto: npm run dev")
        print("2. Abre http://localhost:5173 en el navegador")
        print("3. Abre DevTools (F12) → Network")
        print("4. Recarga la página")
        print("5. Busca el archivo .png del logo")
        print("6. Click derecho → Open in new tab → Guardar")
        print(f"7. Guarda como: {LOGO_PATH}")
        print("\n💡 O usa el método alternativo:")
        print("   - Ve a: https://www.pwabuilder.com/imageGenerator")
        print("   - Sube tu logo y descarga los iconos\n")
        sys.exit(1)

def create_output_dir():
    """Crear directorio de salida si no existe"""
    OUTPUT_DIR.mkdir(parents=True, exist_ok=True)

def generate_icon(image, size, maskable=False):
    """Generar un icono individual"""
    if maskable:
        # Para maskable, agregar padding del 10%
        padding = int(size * 0.1)
        image_size = size - (padding * 2)
        
        # Crear canvas con fondo verde oscuro
        icon = Image.new('RGB', (size, size), color='#1e4620')
        
        # Redimensionar logo y centrarlo con padding
        resized = image.resize((image_size, image_size), Image.Resampling.LANCZOS)
        icon.paste(resized, (padding, padding))
    else:
        # Sin padding para iconos normales
        icon = image.resize((size, size), Image.Resampling.LANCZOS)
    
    return icon

def generate_all_icons():
    """Generar todos los iconos"""
    print("🎨 Generando iconos PWA desde tu logo...\n")
    print(f"📂 Logo: {LOGO_PATH}")
    print(f"📂 Destino: {OUTPUT_DIR}\n")
    
    # Verificaciones
    check_logo()
    create_output_dir()
    
    try:
        # Cargar imagen
        print("⏳ Cargando logo...")
        logo = Image.open(LOGO_PATH)
        
        # Convertir a RGBA si no lo está
        if logo.mode != 'RGBA':
            logo = logo.convert('RGBA')
        
        print(f"✅ Logo cargado: {logo.width}x{logo.height}px")
        
        # Advertencia si el logo es muy pequeño
        if logo.width < 512 or logo.height < 512:
            print(f"⚠️  ADVERTENCIA: El logo es pequeño ({logo.width}x{logo.height}px)")
            print("   Se recomienda un logo de al menos 512x512px para mejor calidad")
        
        print()
        
        # Generar cada icono
        for icon_spec in ICON_SIZES:
            size = icon_spec['size']
            name = icon_spec['name']
            maskable = icon_spec.get('maskable', False)
            
            output_path = OUTPUT_DIR / name
            icon = generate_icon(logo, size, maskable)
            icon.save(output_path, 'PNG', optimize=True)
            
            print(f"✅ Generado: {name} ({size}x{size}px)")
        
        print("\n✨ ¡Todos los iconos generados exitosamente!")
        print("\n📋 Próximos pasos:")
        print("1. Verifica los iconos: ls -lh public/icons/*.png")
        print("2. Verifica el deployment: node scripts/check-deployment-ready.js")
        print("3. Haz commit: git add public/icons/*.png")
        print("4. Push: git push origin main")
        print("5. Deploy en Vercel siguiendo: DEPLOYMENT_INSTRUCTIONS.md\n")
        
    except Exception as error:
        print(f"❌ Error generando iconos: {error}")
        print("\n💡 Alternativa: Usa PWA Builder")
        print("   https://www.pwabuilder.com/imageGenerator\n")
        sys.exit(1)

if __name__ == '__main__':
    generate_all_icons()
